src/screens/Settings/SettingsScreen.tsx

```diff
@@ -0,0 +1,107 @@
+import AsyncStorage from '@react-native-community/async-storage';
+import React, { useState, useEffect } from 'react';
+import { StyleSheet, View } from 'react-native';
+import { NavigationScreenProps } from 'react-navigation';
+
+import { images, icons } from 'app/assets';
+import { Image, ScreenTemplate, Header, ListItem } from 'app/components';
+import { AppStorage, Biometric } from 'app/legacy';
+import { en } from 'app/locale';
+
+import { LabeledSettingsRow } from './LabeledSettingsRow';
+
+export const SettingsScreen = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [isAdvancedOptions, setIsAdvancedOptions] = useState(false);
+  const [biometrics, setBiometrics] = useState({
+    isDeviceBiometricCapable: false,
+    isBiometricsEnabled: false,
+    biometricsType: '',
+  });
+
+  const onAdvancedOptionsChange = async (value: boolean) => {
+    if (value) {
+      await AsyncStorage.setItem(AppStorage.ADVANCED_MODE_ENABLED, '1');
+    } else {
+      await AsyncStorage.removeItem(AppStorage.ADVANCED_MODE_ENABLED);
+    }
+    setIsAdvancedOptions(value);
+  };
+
+  const onFingerprintLoginChange = async (value: boolean) => {
+    if (await Biometric.unlockWithBiometrics()) {
+      biometrics.isBiometricsEnabled = value;
+      await Biometric.setBiometricUseEnabled(value);
+      setBiometrics(biometrics);
+    }
+  };
+
+  useEffect(() => {
+    (async () => {
+      setIsAdvancedOptions(!!(await AsyncStorage.getItem(AppStorage.ADVANCED_MODE_ENABLED)));
+      const isBiometricsEnabled = await Biometric.isBiometricUseEnabled();
+      const isDeviceBiometricCapable = await Biometric.isDeviceBiometricCapable();
+      const biometricsType = (await Biometric.biometricType()) || 'biometrics';
+      setBiometrics({
+        isBiometricsEnabled,
+        isDeviceBiometricCapable,
+        biometricsType,
+      });
+      setIsLoading(false);
+    })();
+  });
+
+  const renderGeneralSettings = () => (
+    <>
+      <ListItem title={en.settings.language} source={icons.languageIcon} />
+      <ListItem title={en.settings.electrumServer} source={icons.dataUsageIcon} />
+      <ListItem
+        title={en.settings.advancedOptions}
+        source={icons.buildIcon}
+        switchValue={isAdvancedOptions}
+        onSwitchValueChange={onAdvancedOptionsChange}
+        iconHeight={21}
+        iconWidth={20}
+      />
+    </>
+  );
+
+  const renderSecuritySettings = () => (
+    <>
+      <ListItem title={en.settings.changePin} source={icons.lockIcon} iconWidth={15} iconHeight={20} />
+      <ListItem
+        title={en.settings.fingerprintLogin}
+        source={icons.fingerprintIcon}
+        switchValue={biometrics.isBiometricsEnabled}
+        onSwitchValueChange={onFingerprintLoginChange}
+        iconWidth={17}
+        iconHeight={19}
+      />
+    </>
+  );
+
+  const renderAboutSettings = () => <ListItem title={en.settings.aboutUs} source={icons.infoIcon} />;
+
+  return isLoading ? (
```

Comment:
you could just move 
```
if (isLoading) return null
```
to line 15

Reply:
Wouldn't that return null always for that screen?

Reply:
Wouldn't that return null always for that screen?